### ISSUE TITLE
Sorting and filter status of columns is not communicated to the screen reader user

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4663]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 4.1.2: Name, role, value

**Test Environment:**
- Test URL: https://clarity.design/documentation/datagrid/
- OS: Windows 10, MacOs 
- User Agents/Assistive Technologies: Firefox 76.0, Safari
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL.
2. Navigate to any one of the datagrids with sort options.
3. Try to sort the result and reset the result.

**Expected result:**
1. Screen reader user should be provided with some confirmation after sorting the data as well as after resetting the sorted result.
2. Sorted columns status should be communicated to the screen reader while navigating through the sorted columns using forward or backward navigation.

**Actual result:**
1. No confirmation provided for sorting or reset.
2. If the user navigates forward to the next column and navigates back to the sorted column, then sorted status is not communicated to the screen reader user. It is communicated only if the sorted column is focused with forward navigation. For e.g. If there are 3 columns A, B and C. B is sorted in descending order and user navigates to column C and comes back to B, sorted descending order sorted status is not announced. But if user navigates from B to A and navigates from A to B, then descending order status is communicated to the screen reader user.

Similary, there is no proper mechanism available to communicate screen reader user about the filter applied or cleared status. Also, screen reader user cannot distinguish columns with filters applied from the one with filters not applied. Sighted users can see that filter icon is highlighted and circled.
